pre-docs/writing_yaml_plans.md

```diff
@@ -399,5 +399,46 @@ This lets you take advantage of the power of Puppet language in the places it's
 When your plans need more sophisticated control flow or error handling beyond running a list of steps in order, it's time to convert them to [Puppet language plans](./writing_plans.html).
 
 
+## Converting YAML Plans to Puppet Plans
+You can convert a YAML plan to a Puppet plan with the `bolt plan convert` command.
 
+```
+bolt plan convert path/to/my/plan.yaml
+```
+
+This takes the path (relative or absolute) to the YAML plan to be converted and prints the converted Puppet plan to stdout.
+
+**NOTE:** Converting a YAML plan might result in a Puppet plan which is syntactically correct, but behaviorally different from its YAML plan. Always manually verify a converted Puppet plan's functionality.
+
+**NOTE** If you convert a YAML plan to puppet and it does *not* have the same behavior as the YAML plan, please [file an issue in our git repo](https://github.com/puppetlabs/bolt/issues).
+
+For example, with this YAML plan:
+```yaml
+# site-modules/mymodule/plans/yamlplan.yaml
+parameters:
+  nodes:
+    type: TargetSpec
+
+steps:
+  - name: run_task
+    task: sample
+    target: $nodes
+    parameters:
+      message: "hello world"
+
+return: $run_task
+```
+
+I can run the following:
+```
+$ bolt plan convert site-modules/mymodule/plans/yamlplan.yaml
+# WARNING: This is an autogenerated plan. It may not behave as expected.
+plan yaml::task(
```

Comment:
Update example output to show accurate plan name (and pick up indentation niceness)
```
cas@cas-ThinkPad-T460p:~/working_dir/bolt/Boltdir$ bolt plan convert site-modules/mymodule/plans/yamlplan.yaml
# WARNING: This is an autogenerated plan. It may not behave as expected.
plan mymodule::yamlplan(
  TargetSpec $nodes
) {
  $run_task = run_task('sample', $nodes, {'message' => "hello world"})

  return $run_task
}
```